lesson22/src/js/main.js

```diff
@@ -0,0 +1,213 @@
+
+        table.appendChild(tbody).appendChild(fragment);
+    }
+}
+
+const sortButtonAttributes = [
+    {state:"asc",src:"../img/asc.svg", alt:"asc-image"},
+    {state:"desc",src:"../img/desc.svg",alt:"desc-image"},
+    {state:"both",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortCategories = ["id", "age"];
+
+const createSortButtons = (columnKey) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtonAttributes){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-${columnKey}SortButton`,
+                'data-state':button.state
+            }
+        });
+
+        button.state !== "both" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+
+    sortButtonsBox.appendChild(fragment);
+    return sortButtonsBox;
+}
+
+const toggleHiddenClassForSortButton = (currentButton,nextButton) => {
+    currentButton.classList.add("hidden");
+    nextButton.classList.remove("hidden");
+}
+
+const updateTableBody = usersData =>  {
+    document.getElementById("js-usersTableBody").remove();
+    renderTableBody(usersData);
+};
+
+const sortUsersData = (category, usersData, state) => {
+    const cloneUsersData = [...usersData];
+
+    switch (state) {
+        case "asc":
+            cloneUsersData.sort((a, b) => a[category] - b[category]);
+            updateTableBody(cloneUsersData);
+            break;
+        case "desc":
+            cloneUsersData.sort((a, b) => b[category] - a[category]);
+            updateTableBody(cloneUsersData);
+            break;
+        default:
+            updateTableBody(usersData);
+    }
+}
+
+const changeState = (currentState) => {
+    switch (currentState) {
+      case "asc":
+        return "desc";
+      case "desc":
+        return "both";
+      default:
+        return "asc";
+    }
+}
+
+const sortTableBody = usersData => {
+    sortCategories.forEach((category) => {
+        const sortButtonsBox = document.getElementById(`js-${category}Buttons-Box`);
+        
+        sortButtonsBox.addEventListener("click", (e) => {
+            if(e.currentTarget === e.target) return;
+
+            const currentState = e.target.getAttribute("data-state");
+            const nextState  = changeState(currentState);
+            const nextButton = document.querySelector(`#js-${category}Buttons-Box [data-state=${nextState}]`);
```

Comment:
IMO
You can also write like this
```suggestion
            const nextButton = sortButtonsBox.querySelector(`[data-state=${nextState}]`);
```

Reply:
Thanks for your comment! 
It's helpful to know how to write smart.
I'll fix it!

Reply:
I fixed.
https://github.com/nakagawamai/js-lesson/pull/24/commits/3ba0ffa812da848dff3969d02ed51bf323b0d0ba